game/scripts/vscripts/items/item_golden_arrow.lua

```diff
@@ -0,0 +1,64 @@
+LinkLuaModifier("modifier_item_golden_arrow", "items/item_golden_arrow.lua", LUA_MODIFIER_MOTION_NONE)
+
+item_golden_arrow = class({
+	GetIntrinsicModifierName = function() return "modifier_item_golden_arrow" end
+})
+
+modifier_item_golden_arrow = class({
+	IsHidden = function() return false end,
+	GetAttributes = function() return MODIFIER_ATTRIBUTE_MULTIPLE end,
+	IsPurgable = function() return false end,
+	RemoveOnDeath = function() return false end,
+})
+
+function modifier_item_golden_arrow:DeclareFunctions()
+	return {
+		MODIFIER_PROPERTY_PREATTACK_BONUS_DAMAGE,
+		MODIFIER_PROPERTY_PHYSICAL_ARMOR_BONUS,
+	}
+end
+
+function modifier_item_golden_arrow:GetModifierPreAttack_BonusDamage()
+	return self:GetAbility():GetSpecialValueFor("bonus_damage")
+end
+
+function modifier_item_golden_arrow:GetModifierPhysicalArmorBonus()
+	return self:GetAbility():GetSpecialValueFor("bonus_armor")
+end
+
+if IsServer() then
+	function item_golden_arrow:OnSpellStart()
+		local caster = self:GetCaster()
+		local stacks = caster:GetModifierStackCount("modifier_item_golden_arrow", self)
+		local coof = stacks*self:GetSpecialValueFor("gold_per_stack")
+		local base_gold = (self:GetSpecialValueFor("base_gold")+(caster:GetLevel()*30))
```

Comment:
What's `caster:GetLevel()*30`? Don't see that thing in item's description. Anyway, don't use magic numbers in code, put them to specials.

Reply:
it is gold that caster loses when die